sklearn/calibration.py

```diff
@@ -1333,9 +1339,15 @@ def from_predictions(
         >>> disp = CalibrationDisplay.from_predictions(y_test, y_prob)
         >>> plt.show()
         """
-        method_name = f"{cls.__name__}.from_estimator"
+        method_name = f"{cls.__name__}.from_predictions"
         check_matplotlib_support(method_name)
 
+        target_type = type_of_target(y_true)
+        if target_type != "binary":
+            raise ValueError(
+                f"The target y is not binary. Got {target_type} type of target."
+            )
+
```

Comment:
Do we need this check here since we do a similar check in `calibration_curve` itself?

https://github.com/scikit-learn/scikit-learn/blob/845b1fac9e8bcfc146e7b38119ce6c500e00a10b/sklearn/calibration.py#L937-L941

Reply:
For raising consistent error messages, I think so.

Reply:
It's either be more consistent or have more computation. I think the `unique` is `O(nlogn)` because it sorts.

 I guess for the binary case, it shouldn't matter too much.